### PR TITLE
[YUNIKORN-805] Fix race condition on the TestMode flag

### PR DESCRIPTION
**Pull request by 0yukali0**
### What is this PR for?
Avoid data race which R/W in same time.
Add getting TestMode in config object with R lock

### What type of PR is it?
* [x] - Bug Fix
* [ ] - Improvement
* [ ] - Feature
* [ ] - Documentation
* [ ] - Hot Fix
* [ ] - Refactoring

### Todos
* [ ] - Task

### What is the Jira issue?
https://issues.apache.org/jira/browse/YUNIKORN-805

### How should this be tested?
make test with Makefile

### Questions:
* [ ] - The licenses files need update.
* [ ] - There is breaking changes for older versions.
* [ ] - It needs documentation.

### CODE REVIEW

File: pkg/shim/scheduler.go [hunk 1]
```diff
@@ -145,6 +146,10 @@ func (ss *KubernetesShim) register(e *fsm.Event) {
 
 func (ss *KubernetesShim) handleSchedulerFailure(e *fsm.Event) {
 	ss.stop()
+	// testmode will be true when mock scheduler intailize
+	if configured := conf.GetSchedulerConf(); !configured.IsTestMode() {
```
Comment (yangwwei) [inline pkg/shim/scheduler.go:150]: this should be 
```
if !conf.GetSchedulerConf().IsTestMode() {
  ...
}
```
conf.GetSchedulerConf() should never be nil.

**Review by yangwwei — CHANGES_REQUESTED** [review]
the change is incompleted, @0yukali0 could u pls find all the direct references of `conf.TestMode` and make sure they all use `conf.IsTestMode()`?

File: pkg/shim/scheduler.go (hunk 1 above)
Comment (0yukali0) [reply pkg/shim/scheduler.go:150]: ok

**Review by yangwwei — APPROVED** [review]
+1

**Pull request merged**